contrib/mesos/pkg/scheduler/podtask/pod_task.go

```diff
@@ -51,32 +52,40 @@ const (
 	Deleted  = FlagType("deleted")
 )
 
+var defaultRoles = []string{"*"}
+
 // A struct that describes a pod task.
 type T struct {
 	ID          string
 	Pod         api.Pod
-	Spec        Spec
+	Spec        *Spec
```

Comment:
And as the spec is full of slices, we would only push down the nil-issue one level. Let's keep this simple with only one simple nil, namely for the Spec.